var/spack/repos/builtin/packages/elfutils/package.py

```diff
@@ -124,7 +124,9 @@ def configure_args(self):
 
         if "+nls" in spec:
             # configure doesn't use LIBS correctly
-            args.append("LDFLAGS=-Wl,--no-as-needed -L%s -lintl" % spec["gettext"].prefix.lib)
+            # Add -lintl if provided by gettext, otherwise libintl is provided by system's glibc:
+            if any("libintl." in filename.split("/")[-1] for filename in spec["gettext"].libs):
+                args.append("LDFLAGS=-Wl,--no-as-needed -L%s -lintl" % spec["gettext"].prefix.lib)
```

Comment:
`-Wl,--no-as-needed` might still be required, don't you think? (It's a little strange that it's only under `+nls` and nothing else, so maybe it's not needed at all, or only needed for ubuntu or other distributions that default to `-Wl,--as-needed`.) You should also take the opportunity to move this addition to the [flag_handler](https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags) method.